File: 100doors.rb
# http://rosettacode.org/wiki/100_doors
start_time = Time.now

def a_hundred_doors
  doors =[]
  100.times do
    doors << 0 # 0 for closed; 1 for open
  end   
  pass = 1
  while pass < 101
    (0..99).each do |index|
      if (index+1) % pass == 0
        if doors[index] == 0 then doors[index] = 1
        elsif doors[index] == 1 then doors[index] = 0
        end
      end
    end
    pass += 1
  end
  doors
end 

doors = a_hundred_doors
open_doors =[]
doors.each_index do |index|
  open_doors << "Door " + (index+1).to_s if doors[index] == 1
end

puts "Doors left open after 100 passes:"
puts open_doors.inspect

puts ((Time.now - start_time).to_f).to_s + "s"

File: 100doors.swift
// http://rosettacode.org/wiki/100_doors

var doors : [Int] = []
for var i = 0; i < 100; i++ {
  doors.append(0)
}

for var pass = 1; pass < 101; pass++ {
  for var index = 0; index < 100; index++ {
    if ((index+1) % pass == 0) {
      if doors[index] == 0 {
        doors[index] = 1
      } else if doors[index] == 1 {
        doors[index] = 0
      }
    }
  }
}

for var index = 0; index < 100; index++ {
  if doors[index] == 1 {
    // door 1 is index 0, et cetera
    println(index+1)
  }
}
